src/runtime/runtime_rp2350.go

```diff
@@ -2,8 +2,40 @@
 
 package runtime
 
+// #include <stdint.h>
+//
+// static void spinlock_lock(void *lock) {
+//     uint32_t _tmp0, _tmp1;
+//     __asm volatile (                                           \
+//     "1:\n"                                                     \
+//     "ldaexb %1, [%2]\n"                                        \
+//     "movs %0, #1\n" /* fill dependency slot */                 \
+//     "cmp %1, #0\n"                                             \
+//     /* Immediately retry if lock is seen to be taken */        \
+//     "bne 1b\n"                                                 \
+//     /* Attempt to claim */                                     \
+//     "strexb %1, %0, [%2]\n"                                    \
+//     "cmp %1, #0\n"                                             \
+//     /* Claim failed due to intervening write, so retry */      \
+//     "bne 1b\n"                                                 \
+//     : "=&r" (_tmp0), "=&r" (_tmp1) : "r" (lock)                \
```

Comment:
Just realized this is incomplete. You also need to clobber flags and memory at least.